### PR TITLE
PostgreSQL Integration improvments

### PR DESCRIPTION
**Pull request by ZoranPandovski**
## Description

This PR:
*  Fixes https://github.com/mindsdb/mindsdb_private/issues/383 by checking if column value is Null or Nan and cast to 0
* Removes postgres_handler from pep8 ignored files and fix all warnings
* Add initial unit test for the handler
* Add missing README for the handler
* Improves logging and code comments
* 

## Type of change

(Please delete options that are not relevant)

- [X] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📢 Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [X] 📄 This change requires a documentation update

## Verification Process

To ensure the changes are working as expected:

 - [X]   Test Location: Use `psql-improvments` branch
 - [ ]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

## Checklist:

- [X] My code follows the style guidelines(PEP 8) of MindsDB.
- [X] I have appropriately commented on my code, especially in complex areas.
- [X] Necessary documentation updates are either made or tracked in issues.
- [X] Relevant unit and integration tests are updated or added.

### CODE REVIEW

File: mindsdb/integrations/handlers/postgres_handler/README.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# PostgreSQL Integration
+
+This documentation describes the integration of MindsDB with PostgreSQL, a powerful, open-source, object-relational database system. The integration allows for advanced SQL functionalities, extending PostgreSQL's capabilities with MindsDB's features.
+
+## Getting Started
+
+### Prerequisites
+
+   1. Ensure that MindsDB and PostgreSQL are installed on your system or you have access to cloud options.
+   2. If running locally install the dependencies as `pip install [postgres]`.
```
Comment (MinuraPunchihewa) [inline mindsdb/integrations/handlers/postgres_handler/README.md:10]: This should be changed to `pip install mindsdb[postgres]`?

**Review by MinuraPunchihewa — COMMENTED** [review]
Hey @ZoranPandovski,
Just a minor comment on the README, otherwise it looks good to me.

File: mindsdb/integrations/handlers/postgres_handler/README.md (hunk 1 above)
Comment (ZoranPandovski) [reply mindsdb/integrations/handlers/postgres_handler/README.md:10]: Oh yes

**Comment by ZoranPandovski** [comment]
@MinuraPunchihewa Done

**Review by MinuraPunchihewa — APPROVED** [review]

**Pull request merged**